src/Reflex/Class.hs

```diff
@@ -271,12 +301,38 @@ fforMaybe = flip fmapMaybe
 ffilter :: FunctorMaybe f => (a -> Bool) -> f a -> f a
 ffilter f = fmapMaybe $ \x -> if f x then Just x else Nothing
 
-instance Reflex t => FunctorMaybe (Event t) where
-  fmapMaybe f = push $ return . f
+-- | Left-biased event union (prefers left event on simultaneous
+-- occurrence).
+instance Reflex t => Alt (Event t) where
+  ev1 <!> ev2 = leftmost [ev1, ev2]
+
+-- | Event intersection (convenient interface to 'coincidence').
+instance Reflex t => Apply (Event t) where
+  evf <.> evx = coincidence (fmap (<$> evx) evf)
+
+-- | Event intersection (convenient interface to 'coincidence').
+instance Reflex t => Bind (Event t) where
+  evx >>- f = coincidence (f <$> evx)
+  join = coincidence
+
+-- | Event of single-occurrence events: @duplicated ev@ is an event that
+-- occurs at the same times as @ev@ carrying events that occur only once
+-- at that particular point in time.
+instance Reflex t => Extend (Event t) where
+  duplicated ev =
+      pushAlways (\_ -> switch <$> hold ev (never <$ ev))
+                 ev
```

Comment:
This seems like a reasonable instance, but is there a reason that `duplicated e = e <$ e` wouldn't also be legitimate?  It would be good to document that if so.